system-contracts/contracts/libraries/SystemContractsCaller.sol

```diff
@@ -114,6 +114,28 @@ library SystemContractsCaller {
                 success := call(msgValueSimulator, callAddr, value, to, farCallAbi, forwardMask, 0)
             }
         }
+        if (success) {
+            assembly {
+                let returndataSize := returndatasize()
```

Comment:
1. `returndataSize` not used.
2. usual low-level `call` would copy the returndata even if the call was not successful, I suggest copying the returndata regardless of `success`

Reply:
done